util/dvsim/BuildOnlyCfg.py

```diff
@@ -0,0 +1,199 @@
+        ignored_wildcards = ["build_mode", "index", "test"]
+        self.__dict__ = find_and_substitute_wildcards(self.__dict__,
+                                                      self.__dict__,
+                                                      ignored_wildcards)
+
+        # Print info
+        log.info("Scratch path for %s: %s", self.name, self.scratch_path)
+
+        # Set directories with links for ease of debug / triage.
+        self.links = {
+            "D": self.scratch_path + "/" + "dispatched",
+            "P": self.scratch_path + "/" + "passed",
+            "F": self.scratch_path + "/" + "failed",
+            "K": self.scratch_path + "/" + "killed"
+        }
+
+        # Use the default build mode for tests that do not specify it
+        if not hasattr(self, "build_mode"):
+            setattr(self, "build_mode", "default")
+
+        self._process_exports()
+
+        # Create objects from raw dicts - build_modes
+        self._create_objects()
+
+        # Post init checks
+        self.__post_init__()
+
+    def __post_init__(self):
+        # Run some post init checks
+        super().__post_init__()
+
+    @staticmethod
+    def create_instance(flow_cfg_file, proj_root, args):
+        '''Create a new instance of this class as with given parameters.
+        '''
+        return BuildOnlyCfg(flow_cfg_file, proj_root, args)
+
+    # Purge the output directories. This operates on self.
+    def _purge(self):
+        if self.scratch_path is not "":
+            try:
+                log.info("Purging scratch path %s", self.scratch_path)
+                os.system("/bin/rm -rf " + self.scratch_path)
+            except IOError:
+                log.error('Failed to purge scratch directory %s',
+                          self.scratch_path)
+
+    def _create_objects(self):
+        # Create build and run modes objects
+        build_modes = Modes.create_modes(BuildModes,
+                                         getattr(self, "build_modes"))
+        setattr(self, "build_modes", build_modes)
+
+        # All defined build modes are being built, h
+        # ence extend all with the global opts.
+        for build_mode in build_modes:
+            build_mode.build_opts.extend(self.build_opts)
+
+    def _print_list(self):
+        for list_item in self.list_items:
+            log.info("---- List of %s in %s ----", list_item, self.name)
+            if hasattr(self, list_item):
+                items = getattr(self, list_item)
+                for item in items:
+                    log.info(item)
+            else:
+                log.error("Item %s does not exist!", list_item)
+
+    def _create_dirs(self):
+        '''Create initial set of directories
+        '''
+        # Invoking system calls has a performance penalty.
+        # Construct a single command line chained with '&&' to invoke
+        # the system call only once, rather than multiple times.
```

Comment:
I'm not really convinced that the penalty we pay for calling `exec()` multiple times is really relevant (vs. spawning a shell and the syscalls then happing by `rm` and `mkdir` to actually do their job).

Reply:
that is probably something @sriyerg could shed some light on since he originally wrote this class (this is also present in `SimCfg.py`).

Reply:
@sriyerg  ping :)?

Reply:
Our scratch paths are on network drives and accessing those on every command separately adds a performance penalty.  This was indeed an outcome of profiling exercise that saved us 20+ seconds of initial processing time when running all IPs and full chip regressions (with 1000s of dirs being created).